airflow/jobs/scheduler_job.py

```diff
@@ -780,7 +788,12 @@ def process_file(self, file_path, zombies, pickle_dags=False, session=None):
         for dag in dagbag.dags.values():
             dag.sync_to_db()
 
-        paused_dag_ids = {dag.dag_id for dag in dagbag.dags.values() if dag.is_paused}
+        paused_dag_ids = (
+            session.query(DagModel.dag_id)
+            .filter(DagModel.is_paused.is_(True))
+            .filter(DagModel.dag_id.in_(dagbag.dag_ids))
+            .all()
+        )
```

Comment:
A few lines after this we call self._process_dags, which the makes this same query again. Is it worth passing it in instead?

The other thing I'm wondering API wise is if this should be encapsulated inside DagBag (something like a `dagbag.paused_dag_ids` method or accessor) - but that may not play very well with the global/long-lived DagBag object the webserver has.

Reply:
Good catch. dags parameters contain only active dags, so we don't have to check it a second time.